tests/testthat/test-sample_grid.R

```diff
@@ -0,0 +1,59 @@
+context("sample_grid_scan")
+
+# Only tests that a grid search can be run
+test_that("sample_grid_scan works", {
+  set.seed(1)
+  
+  # grab the data
+  data <- read.csv(sircovid_file("extdata/example.csv"),
+                   stringsAsFactors = FALSE)
+  
+  
+  # filter and tidy it
+  # Don't start the data until it is after thid date
+  data <- data[data$date > as.Date("2020-02-29"),]
+  
+  # which dates should be set to NA
+  na_dates <- as.Date(c("2020-03-01","2020-03-02","2020-03-03","2020-03-04",
+                        "2020-03-05","2020-03-06","2020-03-07","2020-03-08",
+                        "2020-03-15","2020-03-16","2020-03-19"))
+  data$itu[data$date %in% na_dates] <- NA
+  
+  # bring the deaths back by 2 days as ECDC is out of sync
+  data$deaths[head(seq_len(nrow(data)), -2)] <- tail(data$deaths, -2)
+  data$deaths[tail(seq_len(nrow(data)), 2)] <- NA
+  
+  
+  # Parameters for run
+  min_beta <- 0.10
+  max_beta <- 0.18
+  beta_step <- 0.04
+  first_start_date <- "2020-01-29"
+  last_start_date <- "2020-02-14"
+  day_step <- 6
```

Comment:
Potentially use a lower number of steps if possible, this was a bit slow to test for me

Reply:
done and changed the n_particles down as that will speed this up too